util/scale_run.sh

```diff
@@ -0,0 +1,230 @@
+#!/bin/bash
+
+# Run scale tests for each Standard Ref Arch size on docs page
+#   Ref: https://puppet.com/docs/pe/latest/hardware_requirements.html
+
+
+##########
+# Function Definitions
+##########
+
+# Display usage help
+usage () {
+    echo "$0 [-h] [-d] PE_VERSION -- Run scale testing on specified PE_VERSION"
+    echo
+    echo "where:"
+    echo "            -h  show this help text"
+    echo "            -d  debug mode: prints ENV values and commands to be executed, but does not run anything"
+    echo "    PE_VERSION  in semver notation (e.g 2019.1.0)"
+    echo
+}
+
+# Debug function to validate that bash subshells inherit
+# expected environment variables.
+function echo_env () {
+    echo "============================================
+In subshell for $test $run_type round $i
+cmd=$cmd
+BEAKER_INSTALL_TYPE=$BEAKER_INSTALL_TYPE
+BEAKER_PE_DIR=$BEAKER_PE_DIR
+BEAKER_PE_VER=$BEAKER_PE_VER
+PUPPET_GATLING_SCALE_BASE_INSTANCES=$PUPPET_GATLING_SCALE_BASE_INSTANCES
+PUPPET_GATLING_SCALE_ITERATIONS=$PUPPET_GATLING_SCALE_ITERATIONS
+PUPPET_GATLING_SCALE_INCREMENT=$PUPPET_GATLING_SCALE_INCREMENT
+PUPPET_GATLING_SCALE_SCENARIO=$PUPPET_GATLING_SCALE_SCENARIO
+ABS_AWS_MOM_SIZE=$ABS_AWS_MOM_SIZE
+============================================"
+}
+
+# Link scale results to nginx content location
+function link_results () {
+    # test name is arg1, pe_ver is arg2
+    source=$HOME/gplt/$1/gatling-puppet-load-test/results/scale
+    dest=/usr/share/nginx/html/gplt/scale/$1/$2
+
+    sudo mkdir -p "${dest}"
+    sudo ln -s "${source}" "${dest}"
+}
+
+# Prepare checkout of puppet-gatling-load-test
+function prep_gplt () {
+    cd "$HOME/gplt" || exit 1
+    mkdir "$1"
+    cd "$1" || exit 1
+    git clone git@github.com:puppetlabs/gatling-puppet-load-test
+    cd gatling-puppet-load-test || exit 1
+    bundle install --path vendor/bundle
+}
+
+
+##########
+# Argument checking
+##########
+
+ARGS=()
+while [ $# -gt 0 ]
+do
+    unset OPTIND
+    unset OPTARG
+    while getopts hd  options
+    do
+    case $options in
+        h)  usage
+            exit 1
+            ;;
+        d)  DEBUG=true
+            ;;
+        esac
+   done
+   shift $((OPTIND-1))
+   ARGS+=($1)
+   shift
+done
+
+# Ensure PE_VERSION is provided
+if [ "${#ARGS[@]}" -le "0" ]; then
+    echo "PE_VERSION not provided"
+    echo
+    usage
+    exit 1
+fi
+PE_VERSION=${ARGS[0]}
+
+# Ensure PE_VERSION is semver
+# e.g. 2019.1.1-rc0-207-g4e47830
+# Regex from: https://regexr.com/39s32
+if ! [[ $PE_VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+((-[0-9a-zA-Z-]+)*)?$ ]]; then
+    echo "Supplied PE_VERSION is malformed"
+    echo
+    usage
+    exit 1
+fi
+
+
+##########
+# Main
+##########
+
+tune=false
+# Global settings
+export BEAKER_INSTALL_TYPE=pe
+export BEAKER_PE_DIR=http://enterprise.delivery.puppetlabs.net/archives/releases/2019.1.0
```

Comment:
This should probably use the inputed pe version (if it is a release version).  And if it isn't, the user has to give one... which I don't see a way for them to do since this is hardcoding it.

Reply:
Logic added to:
1. Interpret whether the given PE_VERSION should be a release or a build
1. Check if tarballs exist for that version